clio-server/src/main/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAO.scala

```diff
@@ -78,6 +100,84 @@ trait PersistenceDAO extends LazyLogging {
       )
       logger.debug(s"Wrote document $document to ${written.toUri}")
     }
+
+  /**
+    * Throw or return the `ClioDocument` at `path`.
+    * @param path to a JSON-encoded `ClioDocument``
+    * @tparam D is the type of `ClioDocument` at `path`
+    * @return the `ClioDocument` at `path`
+    */
+  def documentFromPath[D <: ClioDocument: Decoder](path: Path): D = {
+    decode[D](new String(Files.readAllBytes(path))).fold(throw _, identity)
+  }
+
+  /**
+    * @param path in some params `FileSystem`
+    * @return true if `path` is a "params file"
+    */
+  def isParamsFile(path: Path): Boolean = {
+    val s = path.toString
+    val parts = s.split(File.separator)
+    val last = parts.last
+    val isJson = last.endsWith(".json")
+    val hasUuid = last.split('-').length == 5
+    parts.length == 6 && isJson && hasUuid
```

Comment:
And `parts.length` is `path.getNameCount`

Reply:
Good to know ... again.